### ISSUE TITLE
Cannot publish packages from azure-cli docker image

### ISSUE THREAD

**Issue by psandhu79** [Azure/azure-devops-cli-extension#466]
**Describe the bug**
when trying to use the azure-cli docker image cannot publish from folders that are mapped in.

azure-cli                         2.0.58 *
azure-devops                       0.2.0

If I run the following docker command

docker run --rm -e AZURE_DEVOPS_EXT_PAT=$AZURE_DEVOPS_TOKEN -v /directory/folder:/tmp/build build-images.azure-cli /bin/bash -c az devops login ;az artifacts universal publish --feed myfeed --path /tmp/build --version 0.0.7 --name test-cli

It fails with "The path provided is invalid."
If I run the following

docker run --rm -e AZURE_DEVOPS_EXT_PAT=$AZURE_DEVOPS_TOKEN -v /directory/folderi:/tmp/build build-images.azure-cli /bin/bash -c az devops login ;az artifacts universal publish --feed myfeed --path **/bin** --version 0.0.7 --name test-cli

This works not matter what path I try with a mapped in folder it does not work. I have run the docker interactively and can see the folder is mapped in.

**Comment by atbagga** [comment]
@zarenner @mitchdenny  Can someone take a look at this one?

**Comment by geverghe** [comment]
And is this in someway related/duplicate of #217?
+

**Comment by psandhu79** [comment]
Just for info I created the following dockerfile

`FROM microsoft/azure-cli

RUN az extension add --name azure-devops

RUN az devops configure --defaults organization=https://dev.azure.com/myorg`

This creates the image build-images.azure-cli mentioned above

**Comment by mitchdenny** [comment]
Thanks @psandhu79 - we are working on a solution. There are two parts here, we needed to flight a build that works with alpine, and then we also need to merge in a change to the AZ CLI extension itself which passes the distro information to our backend so we can select the right build on behalf of the client.

We've flighted the alpine bits so now its just landing the PR and we should be able to get the basic elements unblocked - then it will be a matter of the AZ DevOps CLI itself getting updated so the new distro logic runs.

We'll leave this open until we are done-done.

**Comment by espoelstra** [comment]
If you are trying to run multiple commands INSIDE the container you usually need to pass the entire set of commands in quotes as your command line interpreter will parse it before they make it into the container otherwise. Your use of `bash -c` is a good start. I haven't played with the container for uploading artifacts yet but I will be shortly, so I'll see if there is a series of commands that works properly.

Side note, you are using the `-v` syntax instead of `--mount` for the volumes, where if the source file doesn't exist Docker CREATES IT on the host, which leads to all sorts of pain. You should use `--mount type=bind,source=sourcepath,target=destpath` or ensure any paths exist beforehand (but `--mount` is much safer).

**Comment by zarenner** [comment]
Please see latest updates in #217. It is now possible to use Universal Packages inside azure-cli docker, although it does currently require adding a couple missing dependencies to the azure-cli image.

**Comment by geverghe** [comment]
@zarenner - can this issue be closed with the updates on #217 ?

**Comment by geverghe** [comment]
Closing this issue. Feel Free to reopen in case it is not fixed.
